crates/cli/src/lib.rs

```diff
@@ -568,3 +603,46 @@ fn build_ext(
 
     bail!("Failed to retrieve extension path from artifact")
 }
+
+// Write content to a given filepath.
+//
+// We may not have write permission but we may have sudo privilege on unix. So
+// we write to a temp file and then try moving it to given filepath, and retry
+// with sudo on unix.
+fn write_to_file(content: String, filepath: &PathBuf) -> anyhow::Result<()> {
+    // write to a temp file
+    let tempf = std::env::temp_dir().join("__tmp_cargo_php");
+    std::fs::write(&tempf, content)?;
+
+    // Now try moving, `rename` will overwrite existing file.
+    if std::fs::rename(&tempf, filepath).is_err() {
+        #[cfg(unix)]
+        {
+            // if not successful, try with sudo on unix.
+            let s = std::process::Command::new("sudo")
+                .arg("mv")
+                .arg(&tempf)
+                .arg(filepath)
+                .status()?;
+            anyhow::ensure!(s.success(), "Falied to write to {filepath:?}");
+        }
+
+        #[cfg(not(unix))]
+        anyhow::bail!("failed to write to {filepath:?}");
+    }
+
+    Ok(())
+}
+
+#[cfg(unix)]
+fn is_root() -> bool {
+    // nix::unistd::Uid::effective().is_root()
+    let uid = unsafe { libc::getuid() };
+    let euid = unsafe { libc::geteuid() };
+
+    match (uid, euid) {
+        (0, 0) => true,
+        (_, 0) => true, // suid set
```

Comment:
The second case always matches if the first one does.

This would be equivalent:

```suggestion
        (_, 0) => true, // suid set
```